src/GoTrueClient.ts

```diff
@@ -888,6 +888,7 @@ export default class GoTrueClient {
         const { data, error } = await this.exchangeCodeForSession(authCode)
         if (error) throw error
         if (!data.session) throw new AuthPKCEGrantCodeExchangeError('No session detected.')
+        window.location.search = ''
```

Comment:
I think we might want to use https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/delete instead to make sure we only remove the code param?

Reply:
@J0 we also should make sure to not cause a page refresh, do based on some research, we might need something like this:

```js
let url = new URL(window.location.href);
url.searchParams.delete('code')
window.history.replaceState(window.history.state, document.title, url.toString())
```